src/mapbox/mapbox.js

```diff
@@ -278,10 +281,20 @@ export default class Mapbox {
     }
   }
 
+  // If canvas size has changed, reads out the new size and returns true
+  _checkForContainerSizeChange() {
+    const {canvas} = this;
+    if (canvas && (this.width !== canvas.clientWidth || this.height !== canvas.clientHeight)) {
+      this.width = canvas.clientWidth;
+      this.height = canvas.clientHeight;
+      return true;
+    }
+    return false;
+  }
+
   // Note: needs to be called after render (e.g. in componentDidUpdate)
   _updateMapSize(oldProps, newProps) {
-    const sizeChanged = oldProps.width !== newProps.width || oldProps.height !== newProps.height;
-    if (sizeChanged) {
+    if (this._checkForContainerSizeChange()) {
```

Comment:
This will not work as it is only called when props change

Reply:
Hmm, yes, should we start an animationLoop?

Reply:
Mapbox has a `trackResize` prop (but no callback that I could find). Wonder if they are also starting an animation loop.

Maybe we should do it this way (`trackResize: true`) starts the loop, since we don't need the loop if deck.gl is managing the size.